frontends/concrete-python/tests/execution/test_bit_extraction.py

```diff
@@ -203,7 +203,144 @@ def test_bit_extraction(input_bit_width, input_is_signed, operation, helpers):
 
         compiler = fhe.Compiler(operation, {"x": "encrypted"})
         circuit = compiler.compile(inputset, helpers.configuration())
-
+        print(circuit.mlir)
         values = inputset if len(inputset) <= 8 else random.sample(inputset, 8)
         for value in values:
             helpers.check_execution(circuit, operation, value, retries=3)
+
+
+def mlir_count_ops(mlir, operation):
+    """
+    Count op in mlir.
+    """
+    return sum(operation in line for line in mlir.splitlines())
+
```

Comment:
maybe adding execution tests for each case

Reply:
I'm sure we have detailed tests for all cases, but feel free to add them to `execution/test_bit_extraction.py` if you want!

Reply:
Some tests added.